kubernetes-sigs/cloud-provider-azure
# [e2e][Windows] Adjust Service validation for Windows cluster

Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>



#### What type of PR is this?
/kind cleanup


#### What this PR does / why we need it:
* Use agnhost image to support Windows Service validation.
  nginx image is only for Linux. Allocate it to control plane node
* Customize deployment's port configuration to support
  more testcases (tcp + udp)
* Replace curl with nc to test UDP traffic
* Unify validate connectivity methods for external & internal IP
* Bug fix: validate connectivity should also consider
  non-80 port, instead of skipping
* Use HostExecPod's IP as LoadBalancerSourceRanges to
  validate rule
* Use 2 deployments & services for shared pls test
* refactor
#### Which issue(s) this PR fixes:

Fixes #705

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:


```docs

```
